system_setup/server/controllers/configure.py

```diff
@@ -127,29 +128,39 @@ async def _install_check_lang_support_packages(self, lang, env) -> bool:
         matches = pattern.match(lang)
         if matches is None:
             log.error("Failed to match expected language format: %s", lang)
-            return False
+            return []
 
         langCodes = matches.groups()
         if len(langCodes) != 1:
             log.error("Only one match was expected for: %s", lang)
-            return False
+            return []
 
         clsLang = langCodes[0]
         # Running that command doesn't require root.
         cp = await arun_command([clsCommand, "-l", clsLang], env=env)
         if cp.returncode != 0:
             log.error('Command "%s" failed with return code %d',
                       cp.args, cp.returncode)
-            return False
+            return []
 
-        packages = [pkg for pkg in cp.stdout.strip().split(' ') if pkg]
-        if len(packages) == 0:
-            log.debug("%s didn't recommend any packages. Nothing to do.",
-                      clsCommand)
-            return True
+        packages = cp.stdout.strip().split(' ')
+        # We will always have language-pack-{clsLang}-base in dryrun.
+        if len(packages) == 0 and self.app.opts.dryrun:
```

Comment:
I like where this is going.  However, if c-l-s fails, we won't hit the packages == [] case, it will actually return an empty list.

Reply:
You're right! Good catch! C-L-S could fail under live-run and be interpreted as recommending nothing, which is not a failure. `None` to the rescue.